### ISSUE TITLE
Killed units not counted

### ISSUE THREAD

**Issue by ccuh** [dcs-liberation/dcs_liberation#780]
[210116mission.zip](https://github.com/Khopa/dcs_liberation/files/5826234/210116mission.zip)

Here is the mission, liberation, and state.json files attached. I reported 2 days ago a problem related to killed units. This is same issue.

The number of the killed units was surely more than 50 or maybe even 100. But results show just 8 other ground units. 0 aircraft, 0 front line units.

Same thing happened again. This time I did not change mission time.

Liberation 2.3.4 (latest) Syria, culling set to 100, full scale war Turkey vs Syria.
![deathofuntrackedgroundunit](https://user-images.githubusercontent.com/77517979/104846189-5bc28b80-58ea-11eb-9fa4-d37c9c049194.png)

**Comment by DanAlbert** [comment]
Not enough information. how did you generate, run, and process the mission results? Did you close the "take off" window ever before submitting the results?

**Comment by Starfire13** [comment]
I have had this problem after I accidentally modified the scenario mission file after the campaign was already generated. Make sure not to do that.

**Comment by HighwaymanEd** [comment]
I second Starfire's comments, don't change anything in the mission. Although it seems that liberation does not count anything that is 'generated' in my experience, i.e. is not a front line unit, or is not a base defense unit.

For example, this is a mission we flew on our dedicated server last night was a SEAD mission against an SA-2 site. The state.json file shows quite a few units destroyed, that when manually imported into the campaign don't seem to fully align at first glance;

`{"base_capture_events":[],"destroyed_objects_positions":[{"orientation":52.001135295164,"type":"T-72B","x":-299199.97414908,"y":576.47960715991,"z":758601.59454558},{"orientation":35.737267456971,"type":"BMP-2","x":-298121.74990857,"y":536.77945993784,"z":759182.20319599},{"orientation":39.804212159235,"type":"BMP-2","x":-298499.91323703,"y":565.91101370902,"z":759211.78389873},{"orientation":10.620259152875,"type":"BMP-2","x":-298486.63888907,"y":563.12789999064,"z":759211.2362229},{"orientation":42.357226188227,"type":"T-55","x":-298103.89154958,"y":536.64620643553,"z":759198.84047264},{"orientation":35.928081909528,"type":"T-55","x":-298121.10773058,"y":536.77378984623,"z":759182.67442165},{"orientation":213.10391435297,"type":"SAU Gvozdika","x":-303074.72212708,"y":649.11647033841,"z":752056.74535605},{"orientation":41.074270723584,"type":"Strela-10M3","x":-308217.78125,"y":1296.3385362838,"z":748193.0625},{"orientation":38.366225321092,"type":"BTR-80","x":-298083.64262315,"y":536.37395058281,"z":759220.22075688},{"orientation":268.19711116835,"type":"ZU-23 Emplacement Closed","x":-287766.59375,"y":166.1993560791,"z":729570.375},{"orientation":80.184171783733,"type":"BMP-1","x":-287400.875,"y":165.76897457287,"z":729364.18749993},{"orientation":292.20074994202,"type":"BMP-2","x":-286445.15625,"y":161.41896737593,"z":728995},{"orientation":38.378254770366,"type":"MTLB","x":-298061.21818452,"y":536.00756974702,"z":759242.96887818},{"orientation":268.19704807649,"type":"ZU-23 Emplacement Closed","x":-287735.59375,"y":166.40895080566,"z":729570.375},{"orientation":268.19736523711,"type":"ZU-23 Emplacement Closed","x":-287766.59375,"y":166.62103271484,"z":729539.375},{"orientation":144.52522981722,"type":"SNR_75V","x":-279542.03125,"y":64.851425170898,"z":685774.375},{"orientation":35.092862298967,"x":-291913.375,"y":23.399900436401,"z":677177.4375},{"orientation":144.52529433482,"type":"p-19 s-125 sr","x":-279562.03125,"y":64.79613494873,"z":685774.375},{"orientation":54.517676878603,"type":"S_75M_Volhov","x":-279503.84375,"y":65.273658752441,"z":685879.375}],"killed_aircrafts":[],"killed_ground_units":["unit|18|23|1002|T-72B| Unit #1","unit|18|21|1002|BMP-2| Unit #1","unit|18|19|1002|BMP-2| Unit #1","unit|18|19|1002|BMP-2| Unit #2","unit|18|28|1002|T-55| Unit #2","unit|18|28|1002|T-55| Unit #1","unit|18|22|1002|SAU Gvozdika| Unit #1","unit|18|27|1002|Strela-10M3| Unit #1","unit|18|31|1002|BTR-80| Unit #1","RED|SAM|110|AAA#2","aa|108|Armor#6","aa|112|Armor#2","unit|18|24|1002|MTLB| Unit #1","RED|SAM|110|AAA#4","RED|SAM|110|AAA#1","RED|SAM|73|TR",87463837,"RED|SAM|73|SR","RED|SAM|73|LN#0"],"mission_ended":true}`

But the actual results showed only 10 units, 2 Frontline, 8 Others;
![image](https://user-images.githubusercontent.com/32423630/104858535-1054bf80-5974-11eb-828e-cae0ed5ab752.png)

And the Frontline units seem to be the only ones that actually count in the debriefing;
![image](https://user-images.githubusercontent.com/32423630/104858542-1fd40880-5974-11eb-9af3-61564923aff0.png)

However, in the campaign itself, the Strike targets killed are counted, such as the 3 x SAM units at the end of the state.json that you can see highlighted below in the campaign manager, that were our actual target;
![image](https://user-images.githubusercontent.com/32423630/104858624-c6200e00-5974-11eb-976c-bf1c21454ab5.png)


In summary, it seems that only units with the name starting with "unit|" are tallied in the mission status scores window, and only front line units are shown in the debriefing window, but liberation is indeed tracking ALL of the units killed.

Maybe a subtle change to the naming of these windows would resolve the confusion, such as;

In the "Waiting for mission completion" window, add a line for "Strategic units" and add the tally for things like SAM launchers, then in the Debriefing window, change it to "Front Line Casualty Reports" to make it clear that these losses are only those from the front lines.

**Comment by ccuh** [comment]
@DanAlbert 
I generated the mission on the server, then took the nextturn.miz file on my pc. Opened it in dcs and save, then put back to server.

In the server I opened the same .liberation file and click takeoff, then start the mission but did not close the window.

When I press the takeoff, that smaller window appears. In the past it didnt matter whether I close it or not but now there is this problem.

By the way after I restarted the server and tried opening the liberation again and submit the json file, I got all results. Strangely it worked. After closing and reopening everything maybe memory cleared and things worked...

Also a wish, it would be very nice to add the possibility to adjust all mission settings like unlimited weapons restrictions, civil traffic etc., weather settings

**Comment by DanAlbert** [comment]
This isn't different from the other bug you filed then. Your workflow does not work in 2.3. The mission is only valid until you close the "take off" window. You're generating a mission, editing it, and then clicking "take off" again. When you click it the second time you break the mission.

This is already fixed in 2.4. There's no point filing additional bugs related to this problem in 2.3.
